test/recipes/15-test_out_option.t

```diff
@@ -0,0 +1,61 @@
+#! /usr/bin/env perl
+# Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use strict;
+use warnings;
+
+use File::Spec;
+use OpenSSL::Test qw/:DEFAULT srctop_file/;
+use OpenSSL::Test::Utils;
+
+setup("test_out_option");
+
+plan skip_all => "'-out' option tests are not available on Windows or VMS"
+    if $^O =~ /^(VMS|MSWin32)$/;
+
+plan tests => 11;
+
+# The following patterns should be tested:
+#
+# path        dirname
+# /usr/       /
+# /           /
+# .           .
+# ..          .
+
+test_illegal_path('/usr/');
+test_illegal_path('/');
+test_illegal_path('.');
+test_illegal_path('..');
```

Comment:
I'm not clear why these are illegal

Reply:
Because they are directories, used directly as is with `-out`.  So this correctly tests that trying to write to a directory (instead of a file in a directory) fails.